### ISSUE TITLE
Export List/Tuple type inputs with dynamic size

### ISSUE THREAD

**Issue by MaxeeCR** [onnx/onnx#5703]
# Ask a Question

### Background

I have a pytorch model with multiple inputs as below:
```
class MyModel():
  ...
  def forward(self, input):
  ""
  input: [object_batch_signals, object_batch_location, object_batch_sequences, agents_batch_signals,  agents_batch_location, agents_batch_category]
  object_batch_signals    (Tensor):         (B, C)
  object_batch_location   (Tensor):         (B, 2)
  object_batch_sequences  (Tensor):         (B, 3, 224, 224)
  agents_batch_signals    (List(Tensor)):   [B, Tensor(num_of_agents, C) or None]
  agents_batch_location   (List(Tensor)):   [B, Tensor(num_of_agents, 2) or None]
  agents_batch_category   (List(Tensor)):   [B, Tensor(num_of_agents, 1) or None]
  """
  ...
```
### About input
'object_batch_signals', 'object_batch_location', 'object_batch_sequences' are **normal `Tensor` type**
'agents_batch_signals',  'agents_batch_location', 'agents_batch_category' are `List` type which **include multiple `Tensor`s with difference shapes**.   

### About axis
* Dynamic axis
  * `B`: batch-size
  * `num_of_agents`: number of objects
* Static axis
  * `C`: channel size of feature 

### Question
Model has been trained successfully. But I don't know how to set inputs' shape when converting **pytorch->onnx->tensorrt**.
```
torch.onnx.export(
  model=model.cpu(),
  args=inputs,
  f=out_onnx_path,
  export_params=True,
  verbose=False,
  opset_version=opset_version,
  input_names=input_names,
  output_names=output_names,
  do_constant_folding=True,
  dynamic_axes=dynamic_axes)   <-----
```

Normal `dynamic_axes` for `Tensor` input should be like: 
```
dynamic_axes = {
  'object_batch_signals': {0: 'batch_size'},
  'object_batch_location': {0: 'batch_size'},
  'object_batch_sequences': {0: 'batch_size'},
  ...
  'output': {0: 'batch_size'}
}
```
But 'agents_batch_signals',  'agents_batch_location', 'agents_batch_category' are `List` type.

Could anyone take a look and help? Thanks!

### Further information
Pytorch version: 2.0
Onnx version: 1.14

**Comment by justinchuby** [comment]
For questions around PyTorch -> ONNX, you are welcome to open the issue in the PyTorch repository, where you will receive more help. Thanks!

**Comment by MaxeeCR** [comment]
Thanks for reply. I will open it in Pytorch repository.
